core/types/gen_header_json.go

```diff
@@ -28,7 +28,7 @@ func (h Header) MarshalJSON() ([]byte, error) {
 		GasLimit         hexutil.Uint64  `json:"gasLimit"         gencodec:"required"`
 		GasUsed          hexutil.Uint64  `json:"gasUsed"          gencodec:"required"`
 		Time             hexutil.Uint64  `json:"timestamp"        gencodec:"required"`
-		Extra            hexutil.Bytes   `json:"extraData"        gencodec:"required"`
+		Extra            hexutil.Bytes   `json:"extraData"        gencodec:"required" deep:"-"`
```

Comment:
Can we avoid the `deep` tags?

Reply:
Agreed, I've removed this and am now checking a few individual fields in the header.